5-Access-Authentication-and-Authorization/5.3-Configure-SSH-Server.bats

```diff
@@ -0,0 +1,212 @@
+    INTERVAL=(${INTERVAL// / })
+    [[ "${INTERVAL[1]}" -gt 1 ]]
+    [[ "${INTERVAL[1]}" -lt 301 ]]
+    local MAX=$(sshd -T -C user=root -C host="$(hostname)" -C addr="$(grep $(hostname) /etc/hosts | awk '{print $1}')" | grep clientalivecountmax)
+    MAX=(${MAX// / })
+    [[ "${MAX[1]}" -lt 4 ]]
+
+    run bash -c "grep -Eis '^\s*clientaliveinterval\s+(0|3[0-9][1-9]|[4-9][0-9][0-9]|[1-9][0-9][0-9][0-9]+|[6-9]m|[1-9][0-9]+m)\b' /etc/ssh/sshd_config /etc/ssh/sshd_config.d/*.conf"
+    [ "$status" -ne 0 ]
+    [[ "$output" == "" ]]
+    run bash -c "grep -Eis '^\s*ClientAliveCountMax\s+(0|[4-9]|[1-9][0-9]+)\b' /etc/ssh/sshd_config /etc/ssh/sshd_config.d/*.conf"
+    [ "$status" -ne 0 ]
+    [[ "$output" == "" ]]
+}
+
+@test "5.3.17 Ensure SSH LoginGraceTime is set to one minute or less (Automated)" {
+    local LOGINGRACETIME=$(sshd -T -C user=root -C host="$(hostname)" -C addr="$(grep $(hostname) /etc/hosts | awk '{print $1}')" | grep logingracetime)
+    LOGINGRACETIME=(${LOGINGRACETIME// / })
+    [[ "${LOGINGRACETIME[1]}" -gt 0 ]]
+    [[ "${LOGINGRACETIME[1]}" -lt 61 ]]
+    run bash -c "grep -Eis '^\s*LoginGraceTime\s+(0|6[1-9]|[7-9][0-9]|[1-9][0-9][0-9]+|[^1]m)' /etc/ssh/sshd_config /etc/ssh/sshd_config.d/*.conf"
+    [ "$status" -ne 0 ]
+    [[ "$output" == "" ]]
+}
+
+@test "5.3.18 Ensure SSH warning banner is configured (Automated)" {
+    run bash -c "sshd -T -C user=root -C host=\"$(hostname)\" -C addr=\"$(grep $(hostname) /etc/hosts | awk '{print $1}')\" | grep banner"
+    [ "$status" -eq 0 ]
+    [[ "$output" == "banner /etc/issue.net" ]]
+    run bash -c "grep -Eis '^\s*Banner\s+\"?none\b' /etc/ssh/sshd_config /etc/ssh/sshd_config.d/*.conf"
+    [ "$status" -ne 0 ]
+    [[ "$output" == "" ]]
+}
+
+@test "5.3.19 Ensure SSH PAM is enabled (Automated)" {
+    run bash -c "sshd -T -C user=root -C host=\"$(hostname)\" -C addr=\"$(grep $(hostname) /etc/hosts | awk '{print $1}')\" | grep -i usepam"
+    [ "$status" -eq 0 ]
+    [[ "$output" == "usepam yes" ]]
+    run bash -c "grep -Eis '^\s*UsePAM\s+no' /etc/ssh/sshd_config /etc/ssh/sshd_config.d/*.conf"
+    [ "$status" -ne 0 ]
+    [[ "$output" == "" ]]
+}
+
+@test "5.3.20 Ensure SSH AllowTcpForwarding is disabled (Automated)" {
+    run bash -c "sshd -T -C user=root -C host=\"$(hostname)\" -C addr=\"$(grep $(hostname) /etc/hosts | awk '{print $1}')\" | grep -i allowtcpforwarding"
+    [ "$status" -eq 0 ]
+    [[ "$output" == "allowtcpforwarding no" ]]
+    run bash -c "grep -Eis '^\s*AllowTcpForwarding\s+yes\b' /etc/ssh/sshd_config /etc/ssh/sshd_config.d/*.conf"
+    [ "$status" -ne 0 ]
+    [[ "$output" == "" ]]
+}
+
+@test "5.3.21 Ensure SSH MaxStartups is configured (Automated)" {
+    run bash -c "sshd -T -C user=root -C host=\"$(hostname)\" -C addr=\"$(grep $(hostname) /etc/hosts | awk '{print $1}')\" | grep -i maxstartups"
+    [ "$status" -eq 0 ]
```

Comment:
This only checks that the `maxstartups` entry exists, but not that it is `10:30:60 or more restrictive`.